src/Sylius/Bundle/LocaleBundle/Doctrine/EventListener/LocaleModificationListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\LocaleBundle\Doctrine\EventListener;
+
+use Doctrine\ORM\Event\PostPersistEventArgs;
+use Doctrine\ORM\Event\PostRemoveEventArgs;
+use Doctrine\ORM\Event\PostUpdateEventArgs;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\Component\Locale\Provider\CachedLocaleCollectionProvider;
+use Symfony\Contracts\Cache\CacheInterface;
+
+final class LocaleModificationListener
+{
+    public function __construct(private CacheInterface $cache)
+    {
+    }
+
+    public function postRemove(PostRemoveEventArgs $event): void
+    {
+        $this->onLocaleChanged($event);
+    }
+
+    public function postUpdate(PostUpdateEventArgs $event): void
+    {
+        $this->onLocaleChanged($event);
+    }
+
+    public function postPersist(PostPersistEventArgs $event): void
+    {
+        $this->onLocaleChanged($event);
+    }
+
+    public function onLocaleChanged(PostPersistEventArgs|PostUpdateEventArgs|PostRemoveEventArgs $event): void
```

Comment:
```suggestion
    private function invalidateCache(PostPersistEventArgs|PostUpdateEventArgs|PostRemoveEventArgs $event): void
```
or alternatively, leave only this method and configure it directly in the service tags